src/plugins/intel_gpu/src/plugin/ops/experimental_detectron_prior_grid_generator.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <intel_gpu/plugin/program.hpp>
+#include <intel_gpu/plugin/common_utils.hpp>
+
+#include <intel_gpu/primitives/experimental_detectron_prior_grid_generator.hpp>
+#include <ngraph/op/experimental_detectron_prior_grid_generator.hpp>
+
+namespace ov {
+namespace runtime {
+namespace intel_gpu {
+namespace {
+cldnn::tensor mkTensor(const ov::Shape &shape) {
+    if (shape.size() == 4)
+        return cldnn::tensor{cldnn::batch(shape[0]), cldnn::feature(shape[1]), cldnn::spatial(4, shape[2])};
+    else
+        return cldnn::tensor{cldnn::spatial(4, shape[0])};
+}
+
+static void CreateExperimentalDetectronPriorGridGeneratorOp(Program &p, const std::shared_ptr<ngraph::op::v6::ExperimentalDetectronPriorGridGenerator> &op) {
+    p.ValidateInputs(op, { 3 });
+    cldnn::tensor outTensor = mkTensor(op->get_output_shape(0));
+    auto outDataType = DataTypeFromPrecision(op->get_output_element_type(0));
+    cldnn::layout outLayout { outDataType, cldnn::format::bfyx, outTensor };
+    auto &attrs = op->get_attrs();
+    auto &priors_shape = op->get_input_shape(0);
+    auto &featmap_shape = op->get_input_shape(1);
+    auto &image_shape = op->get_input_shape(2);
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    inputPrimitives.resize(1); // only priors is read
+    cldnn::experimental_detectron_prior_grid_generator prim {
+        layer_type_name_ID(op),
+        inputPrimitives,
+        outLayout,
+        attrs.flatten,
+        attrs.h,
+        attrs.w,
+        attrs.stride_x,
+        attrs.stride_y,
+        featmap_shape[2],
+        featmap_shape[3],
```

Comment:
maybe would be more readable to send shapes themselves to constructor, and what value to extract will be impl detailses?
Constructors with more than 10 params look little bit horrible...

Reply:
i can easily pass shapes as vectors, but most arguments are taken by attributes. to pass attributes as struct i'd have to include ngraph/op/experimental_detectron_prior_grid_generator.hpp in src/plugins/intel_gpu/include/intel_gpu/primitives/experimental_detectron_prior_grid_generator.hpp
it'd look like layering violation, i didn't see other ops doing it

Reply:
i can create new struct for all constructor arguments and populate it with named initializers if you'd like

Reply:
new struct to init the struct? Looks like design recursion :)

Reply:
for attributes it's ok to send as params, for shapes - I think it's better to send it as shapes - so primitive would encapsulate what exact info from shape is needed for it

Reply:
struct without constructor to do named initialization. it can't be done with `experimental_detectron_prior_grid_generator` because it has to call base constructor. primitive doesn't know how shapes are encoded in ngraph, this file is boundary between ngraph and primitive and it should do conversion. i.e. even if we reduce number of arguments(which i agree is worthwhile goal), conversion still should happen in this file

Reply:
I see that shape is sent to a lot of primitives as tensor or ov::Shape (for example strided_slice, gather_elements, extract_image_patches, etc)

Reply:
we are probably looking at different sources